internal/handlers/common/update_array_operators.go

```diff
@@ -91,6 +96,44 @@ func processPopArrayUpdateExpression(doc *types.Document, update *types.Document
 	return changed, nil
 }
 
+// checkUnsuitableValueError returns ErrUnsuitableValueType when part of
+// the path is used by non document value type.
+// If path is "v.foo" and
+// - doc is {v: 42} it returns ErrUnsuitableValueType, v is used by unsuitable value type.
+// - doc is {c: 10} it returns no error since the path does not exist.
+func checkUnsuitableValueError(doc *types.Document, key string, path types.Path) error {
```

Comment:
Thanks for experimenting. I also attempted, but came to the conclusion that `getPath` is sufficiently different. Because `getPath` expects to return error if it does not exist, on the other hand the user case for this is no error if  the path doesn't exist